### ISSUE TITLE
FF 67 Update Homepage link not works

### ISSUE THREAD

**Issue by Woit** [MatMoul/quickdial-webext#139]
Mac OS, Firefox 67.0.4.
When im trying to set QuickDial URL page as a start page in settings (something like "moz-extension://6985a784-9a68-3f40-b473-40936c7adf35/dial") it not works - firefox opening self main page on any new window. Issue only for new window, if i try to open new tab in exists window - quickdial opens normally

**Comment by MatMoul** [comment]
@Woit Have you looked the video https://youtu.be/zBmdZ4d4Yz4 @ 1:28 ?

**Comment by Woit** [comment]
> @Woit Have you looked the video https://youtu.be/zBmdZ4d4Yz4 @ 1:28 ?

Yep, extension link should replace standart firefox homepage in my case

**Comment by MatMoul** [comment]
Try to edit setting of QD (right click, Quick Dial Settings, at the bottom, try to set "Startpage timeout" to 0 or try to increase the value.

**Comment by Woit** [comment]
no effect (

**Comment by BrianDeClercq** [comment]
Hello.
I am having the same problem, I am not able to set Quick Dial as home page.
The "Use Current Page" button fails to fill in the QD extension address.
The field remains blank as if the quick dial tab was not open. 
I have also seen it switch back to "Firefox Home(default)" as soon as clicking on "Use Current Page".

Changing startpage timeout had no effect.

I am running FF 68.0 on Ubuntu 18.04. 

Thanks, this extension is appreciated, I've been using it for years.

**Comment by MatMoul** [comment]
@BrianDeClercq You say you use this extension for several years, can you tell me which version added the problem?

**Comment by BrianDeClercq** [comment]
First, I can confirm this issue began with update to 68.0.
I just tested a FF 67.0.4 in a Windows 10 machine and found it to work normally. As soon as I allowed it to update to 68.0 and reopened FF, the homepage was Firefox default and I am unable to set it back.

Secondly, I have found the issue--Private Browsing mode.

I have my FF preferences set to "Always use private browsing mode".
I also have set Quickdial as "Allowed in private windows".
Despite this, it is prevented from functioning as home page while in private mode.
Upon disabling "Always use private browsing mode", I am immediately able to set and use Quickdial as homepage. And when re-enabling private mode, Quickdial is instantly removed from the homepage setting and is not allowed to be selected.

Quickdial does work properly when opening new tabs while in private mode. It is only the home page functionality that breaks.

I hope this helps, and thanks again!

**Comment by MatMoul** [comment]
@BrianDeClercq Thanks for your feedback, I suspect that it is the "Startpage timeout" who need to be adjusted.

I explain, when you use QD as start page, QD need to be loaded and it can take time.
If QD is not fully loaded when the start page need to be display, a blank page is displayed.
With the "Startpage timeout", you can set a value that replace "about:blank" page with QD after the timeout.

Try to increase "Startpage timeout".
An other possibilities is that FF68 use an other url and this requiere an update.
